eth_tester/backends/pyevm/main.py

```diff
@@ -627,21 +632,18 @@ def _normalize_transaction(self, transaction, block_number="latest"):
             or
             # if no fee params exist, default to dynamic fee transaction:
             not any(
-                _ in transaction
-                for _ in DYNAMIC_FEE_TRANSACTION_PARAMS + ("gas_price",)
+                _ in transaction for _ in DYNAMIC_FEE_TRANSACTION_PARAMS + ("gasPrice",)
             )
         )
-        is_typed_transaction = (
-            is_dynamic_fee_transaction or "access_list" in transaction
-        )
+        is_typed_transaction = is_dynamic_fee_transaction or "accessList" in transaction
 
         for key in transaction:
             if key in ("from", "type"):
                 continue
             if key == "v" and is_typed_transaction:
-                yield "y_parity", transaction[
+                yield "yParity", transaction[
```

Comment:
I can be convinced to ignore this here since we're going to experiment with the pydantic models, but part of normalization should be camelCase -> snakeCase. Instead of "normalizing" and then calling `dict_keys_to_snake_case` randomly inside `_create_type_aware_unsigned_transaction`, there should be a separation of logic where that doesn't need to be coerced into snake case outside of the normalization process.

Normalization is done to normalize the object for the backend. If the backend is expecting snake-case it should be handled in that step.